pkgs/applications/science/material-science/quantum-espresso/default.nix

```diff
@@ -0,0 +1,57 @@
+{ bash
+, stdenv
+, fetchurl
+, gfortran
+, fftw
+, blas
+, liblapack
+, mpiSupport ? false, mpi ? null
```

Comment:
Shouldn't `mpi ? null` be sufficient? You could the check if mpi is provided with `mpi != null` in the conditionals further down. Since `openmpi` is the default you could even consider replacing `mpi ? null` with just  `openmpi`.